src/main/java/org/jabref/model/strings/StringUtil.java

```diff
@@ -757,4 +763,43 @@ public static String removeStringAtTheEnd(String string, String stringToBeRemove
     public static boolean endsWithIgnoreCase(String string, String suffix) {
         return StringUtils.endsWithIgnoreCase(string, suffix);
     }
+
+    /**
+     * Normalizes a string by standardizing whitespace and punctuation. This includes:
+     * - Trimming outer whitespace
+     * - Converting multiple whitespace characters to a single space
+     * - Converting line breaks to spaces
+     * - Standardizing formatting around punctuation (hyphens, commas, semicolons, colons)
+     *
+     * @param value The string to normalize
+     * @return The normalized string, or empty string if input is null
+     */
+    public static String normalize(String value) {
+        if (value == null) {
+            return "";
+        }
+
+        String withoutLineBreaks = LINE_BREAKS.matcher(value).replaceAll(" ");
+
+        String trimmed = withoutLineBreaks.trim();
+        return NORMALIZE_PATTERN.matcher(trimmed).replaceAll(match -> {
+            String matchStr = match.group();
+            if (matchStr.matches("\\s+")) {
+                return " ";
+            }
+            if (matchStr.matches("\\s*-+\\s*")) {
+                return "-";
+            }
+            if (matchStr.matches("\\s*,\\s*")) {
+                return ", ";
+            }
+            if (matchStr.matches("\\s*;\\s*")) {
+                return "; ";
+            }
+            if (matchStr.matches("\\s*:\\s*")) {
+                return ": ";
+            }
```

Comment:
This needs to be done with named groups - otherwise, we don't have the bvenefit of `Pattern.compile`